python/dask_cudf/dask_cudf/core.py

```diff
@@ -338,12 +328,30 @@ def repartition(self, *args, **kwargs):
         return super().repartition(*args, **kwargs)
 
     @_dask_cudf_nvtx_annotate
-    def shuffle(self, *args, **kwargs):
+    def shuffle(self, *args, shuffle="tasks", **kwargs):
         """Wraps dask.dataframe DataFrame.shuffle method"""
-        shuffle_arg = kwargs.pop("shuffle", None)
-        if shuffle_arg and shuffle_arg != "tasks":
-            raise ValueError("dask_cudf does not support disk-based shuffle.")
-        return super().shuffle(*args, shuffle="tasks", **kwargs)
+        if (
+            shuffle == "explicit-comms"
+            or dask.config.get("explicit-comms", False)
```

Comment:
Yeah, this is a good question. For most cases, we only need the explicit-comms version of `rearrange_by_column` to capture the benefits of an explicit-comms shuffle. The only case were this is "not enough" is the case where the number of output partitions should be less than the number of input partitions. This is because the full explicit-comms shuffle algorithm will perform repartitioning on each worker, while the upstream `shuffle` API will add an extra `ddf.repartition` call **before** calling `rearrange_by_column` will a smaller number of partitions.

Reply:
Ah, good catch - I hadn't realized that the repartitioning was in `rearrange_by_columns` (I thought it was in `shuffle` for some reason).